crates/nu-parser/src/type_check.rs

```diff
@@ -9,22 +9,19 @@ use nu_protocol::{
 
 pub fn type_compatible(lhs: &Type, rhs: &Type) -> bool {
     // Structural subtyping
-    let is_compatible = |expected: &[(String, Type)], found: &[(String, Type)]| {
-        if expected.is_empty() || found.is_empty() {
-            // We treat an incoming empty table/record type as compatible for typechecking purposes
-            // It is the responsibility of the runtime to reject if necessary
-            true
-        } else if expected.len() > found.len() {
-            false
-        } else {
-            expected.iter().all(|(col_x, ty_x)| {
-                if let Some((_, ty_y)) = found.iter().find(|(col_y, _)| col_x == col_y) {
-                    type_compatible(ty_x, ty_y)
-                } else {
-                    false
-                }
+    let is_compatible = |expected: &[(String, Type)], found: &[(String, Type)]| -> bool {
+        // We treat an incoming empty table/record type as compatible for typechecking purposes
+        // It is the responsibility of the runtime to reject if necessary
+        found.is_empty()
+            || expected.is_empty()
+            || expected.len() <= found.len()
```

Comment:
That does not make sense as an `||` condition because it would say erroneously that `table<a, b, c> <: table<d, e>`